### PR TITLE
Implement standard API endpoint: GetStateValidatorBalances

### PR DESCRIPTION
**Pull request by cemozerr**
## PR Description
Implement standard API endpoint: GetStateValidatorBalances. Builds on #2985

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFinalityCheckpoints.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+      }
+      SafeFuture<Optional<BeaconState>> future = chainDataProvider.getStateAtSlot(maybeSlot.get());
+      handleOptionalResult(ctx, future, this::handleResult, SC_NOT_FOUND);
+    } catch (ChainDataUnavailableException ex) {
+      LOG.trace(ex);
+      ctx.status(SC_SERVICE_UNAVAILABLE);
+    } catch (IllegalArgumentException ex) {
+      LOG.trace(ex);
+      ctx.status(SC_BAD_REQUEST);
+      ctx.result(jsonProvider.objectToJSON(new BadRequest(ex.getMessage())));
```
Comment (rolfyone) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFinalityCheckpoints.java]: there's a helper, `ctx.result(BadRequest.badRequest(jsonProvider, ex.getMessage))`

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFork.java [hunk 2]
```diff
@@ -71,8 +71,7 @@ public GetStateFork(final DataProvider dataProvider, final JsonProvider jsonProv
       method = HttpMethod.GET,
       summary = "Get state fork",
       tags = {TAG_V1_BEACON, TAG_VALIDATOR_REQUIRED},
-      description =
-          "Request beacon node to provide all validators that are scheduled to propose a block in the given epoch.",
+      description = "Returns Fork object for state with given 'stateId'.",
```
Comment (rolfyone) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFork.java]: `state_id`

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFinalityCheckpoints.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+      method = HttpMethod.GET,
+      summary = "Get state finality checkpoints",
+      tags = {TAG_V1_BEACON},
+      description =
+          "Returns finality checkpoints for state with given 'stateId'. In case finality is not yet achieved, checkpoint should return epoch 0 and ZERO_HASH as root.",
+      pathParams = {@OpenApiParam(name = PARAM_STATE_ID, description = PARAM_STATE_ID_DESCRIPTION)},
+      responses = {
+        @OpenApiResponse(
+            status = RES_OK,
+            content = @OpenApiContent(from = GetStateRootResponse.class)),
```
Comment (rolfyone) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFinalityCheckpoints.java]: probably going to be a different response object here?

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFinalityCheckpoints.java (hunk 1 above)
Comment (rolfyone) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFinalityCheckpoints.java]: looks like `finalityCheckpointsResponse`

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFork.java (hunk 2 above)
Comment (cemozerr) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFork.java]: Done.

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFinalityCheckpoints.java (hunk 1 above)
Comment (cemozerr) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFinalityCheckpoints.java]: Done.

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFork.java (hunk 2 above)
Comment (cemozerr) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFork.java]: Done.

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFinalityCheckpoints.java (hunk 3 above)
Comment (cemozerr) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFinalityCheckpoints.java]: Fixed.

**Review by rolfyone — APPROVED** [review]
LGTM

**Comment by cemozerr** [comment]
Closing this in favor of the super-set PR: https://github.com/PegaSysEng/teku/pull/3003

**Pull request closed without merging**